centreon/www/include/monitoring/status/Services/xml/serviceXML.php

```diff
@@ -165,45 +139,107 @@
     h.icon_image AS h_icon_images, h.display_name AS h_display_name, h.action_url AS h_action_url,
     h.notes_url AS h_notes_url, h.notes AS h_notes, h.address,
     h.passive_checks AS h_passive_checks, h.active_checks AS h_active_checks,
-    i.name as instance_name, cv.value as criticality, cv.value IS NULL as isnull
-    FROM hosts h, instances i ";
+    i.name as instance_name, cv.value as criticality, cv.value IS NULL as isnull,
+    m.index_id IS NOT NULL as hasGraph
+    FROM hosts h
+    INNER JOIN instances i
+      ON h.instance_id = i.instance_id
+    INNER JOIN services s
+      ON s.host_id = h.host_id 
+    SQL;
+
+if (!$obj->is_admin) {
+    $request .= <<<SQL
+        INNER JOIN centreon_acl
+            ON centreon_acl.host_id = h.host_id
+            AND centreon_acl.service_id = s.service_id
+            AND group_id IN ({$obj->grouplistStr})
+        SQL;
+}
+
 if (isset($hostgroups) && $hostgroups != 0) {
-    $request .= ", hosts_hostgroups hg, hostgroups hg2";
+    $request .= <<<SQL
+        INNER JOIN hosts_hostgroups hg
+            ON hg.host_id = h.host_id
+            AND hg.hostgroup_id = :hostGroup
+        INNER JOIN hostgroups hg2
+            ON hg2.hostgroup_id = hg.hostgroup_id
+        SQL;
+
+    $queryValues['hostGroup'] = [\PDO::PARAM_INT => $hostgroups];
```

Comment:
It was already like that, but it's weird to have name `$hostgroups` which is a hostgroup id